### ISSUE TITLE
Prevent undefined link to explorer for Bitcoin transaction on submitted screen

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#4446]
https://mempool.space/tx/undefined

**Comment by markmhendrickson** [comment]
This is a problem on the submission modal even after waiting seconds after broadcast:



![Image](https://github.com/leather-wallet/extension/assets/28991/eb4072c1-c19a-4a4e-a60d-5940a59e0d23)



![Image](https://github.com/leather-wallet/extension/assets/28991/361f485e-49cf-4cf7-b1ed-2ddaef91b117)

**Comment by pete-watters** [comment]
@markmhendrickson what do you want to happen here? 

Should I keep the button disabled if the `txId` is undefined?

**Comment by markmhendrickson** [comment]
Ideally the button is disabled until the txId is defined, though it seems like in my testing it's _never_ defined even if I wait for awhile on this screen. So, this issue could involve:

1) Disabling button while loading txID
2) Fixing issue where txID is never loaded

**Comment by pete-watters** [comment]
I just tested this as I had to send some funds and I noticed that when you click `View Details` it does indeed not work, however if you click `Copy ID` the ID is already available and you can use that to find the transaction in MemPool explorer. 



https://github.com/leather-wallet/extension/assets/2938440/5432bea6-73b1-414a-bca1-203d346f9e2f

TLDR - I don't think it will be too tricky to fix

**Comment by pete-watters** [comment]
This bug was introduced in https://github.com/leather-wallet/extension/pull/4349/ as we made some changes in the `useExplorerLink` to change the name of `txid` to `txId`
